gnovm/pkg/gnolang/type_check.go

```diff
@@ -1276,3 +1286,192 @@ func isBlankIdentifier(x Expr) bool {
 	}
 	return false
 }
+
+func getExpr(base Expr, expr Expr, seen *seen[Expr]) (res []Expr) {
```

Comment:
not sure why Gno needs to do this, and why Go's typechecker can't just pick up on it, but it seems this name should be changed, as it does not return a single expression, and getExprs() still isn't descriptive.

I'm also wondering if this could be implemented more succinctly using Transcribe().
See findGotoLoopDefines or findLoopUses1 for example usage, see usage of TRANS_SKIP in findGotoLoopDefines for example, to skip over function bodies.

Reply:
Nice suggestion ^^! I was looking into this as well and couldn’t find anything existing that did it the way I needed. Indeed, using `Transcribe()` is a better fit for this task, it simplifies the implementation. I've updated the code accordingly. Check out the commit here: [[875c04f5](https://github.com/gnolang/gno/pull/3714/commits/875c04f5bced57b6fb2d19f291c8f04b50053a1f)]. 
Thanks again for the feedback ^^!